### PR TITLE
[DI-6248] Update event documentation

### PR DESCRIPTION
**Pull request by isturdy**
# Pull Request/Issue Resolution

#### Description of Change:
This updates the event payload documentation for Currents with one automatically generated from the event config. Aside from fixing some inconsistencies, this allows accurate event structures for all destinations, not merely storage ones.

#### Is this change associated with a Braze feature/product release?
- [ ] Yes (**Insert Feature Release Date Here**)
- [x] No

---

<details>
<summary>✔️ Pull Request Checklist</summary>
<br>

- [x] Check that you haven't removed any images (replacing an image with an updated one of the same name is fine), as this breaks the French site
- [x] Check that all links work.
- [ ] Ensure you have completed [our Contributors License Agreement](https://www.braze.com/docs/cla/).
- [ ] Tag @josh-mccrowell-braze and @internetisaiah as a reviewer when your work is **done and ready to be reviewed for merge**. Are you an internal product manager? Reference the internal reviewing chart to tag the appropriate reviewer.
- [x] If the documentation involves a 1) paid SKU, 2) a third party, 3) SMS, 4) AI, or 5) privacy, ensure that Ana Teresa Serafino on the Legal team has signed off.
- [ ] Tag others as reviewers as necessary.
- [x] If you have modified any links, be sure to add redirects to `assets` > `js` > `broken_redirect_list.js`

</details>

<details>
<summary>⭐ Helpful Wiki Shortcuts</summary>
<br>

- [Writing Style Guide](https://docs.google.com/document/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub)
- [Image Style Guide](https://docs.google.com/document/d/e/2PACX-1vRJSkwcjmjrTfLDagZccLpOMMyh5NN5SXRZSjz12cRAHbX4OrUmhvCmYpf_p5YB-9r4_jSOQLkicQIH/pub)
- [Styling Test Page](https://www.braze.com/docs/home/styling_test_page/)

</details>

<details>
<summary>❗ ATTN: For PR Reviewers</summary>
<br>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Read our [Previewing Documentation page](https://github.com/braze-inc/braze-docs/wiki/Previewing-and-Testing-Documentation) to see how to check the deployment.
  - [ ] Preview all changes in the linked Vercel environment by clicking the preview link in the vercel-bot comment in your PR.
</details>

<details>
<summary>❗ ATTN: Internal Reviewing Chart </summary>
<br>
<b>Work at Braze and not sure who to tag for review?</b> <br>Before tagging @josh-mccrowell-braze or @internetisaiah for a general review, reference the following chart to see if a specific product vertical/reviewer applies to your pull request.
<br><br>
<table>
<tr>
    <td><b>Reviewer</b></td>
    <td><b>Product Vertical</b></td>
  </tr>
  <tr>
    <td>@josh-mccrowell-braze</td>
    <td>Monolith Deployments<br>Quality Infrastructure<br>Platform Infrastructure<br>Datalake<br>SDKs<br>Currents</td>
  </tr>
  <tr>
    <td>@internetisaiah</td>
    <td>Developer Guide<br>Technology partners<br>Channels</td>
  </tr>
  <tr>
    <td>@bre-fitzgerald</td>
    <td>Intelligence<br>In-App Messages<br>Channels<br>Frontend Infrastructure & Experience (FIX)</td>
  </tr>
  <tr>
    <td>@lydia-xie</td>
    <td>Ingestion<br>Core Objects<br>Core Messaging<br>Messaging Experience<br>Message Components<br>Email (Composition and Infrastructure) (tag @rachel-feinberg for Liquid use cases)</td>
  </tr>
  <tr>
    <td>@rachel-feinberg</td>
    <td>Customer Lifecycle, Identity and Permissions<br>SMS<br>User Targeting<br>Reporting</td>
  </tr>
</table>
</details>

### CODE REVIEW

**Comment by isturdy** [comment]
@cla-bot check

File: _docs/_user_guide/data_and_analytics/braze_currents/event_glossary/message_engagement_events.md [hunk 1]
```diff
@@ -119,1786 +70,9439 @@ WhatsApp, Delivery
...
+  "from_phone_number" : "(required, string) Phone number used to send in e164 format (for example +14155552671)",
+  "id" : "(required, string) Globally unique ID for this event",
+  "message_variation_id" : "(optional, string) API ID of the message variation this user received",
+  "provider_error_code" : "(required, string) Error code from WhatsApp",
+  "provider_error_title" : "(required, string) Description of failure from WhatsApp",
+  "subscription_group_id" : "(optional, string) Subscription group API ID",
+  "time" : "(required, long) Unix timestamp at which the event happened",
+  "timezone" : "(optional, string) Time zone of the user",
+  "to_phone_number" : "(optional, string) Phone number of the user receiving the message in e164 format (for example +14155552671)",
+  "user_id" : "(required, string) BSON ID of the user that performed this event"
```
Comment (lydia-xie) [inline _docs/_user_guide/data_and_analytics/braze_currents/event_glossary/message_engagement_events.md:368]:
```suggestion
  "app_group_id" : "(optional, string) API ID of the app group this user belongs to",
  "campaign_id" : "(optional, string) API ID of the campaign this event belongs to",
  "canvas_id" : "(optional, string) API ID of the Canvas this event belongs to",
  "canvas_step_id" : "(optional, string) API ID of the Canvas step this event belongs to",
  "canvas_step_message_variation_id" : "(optional, string) API ID of the Canvas step message variation this user received",
  "canvas_variation_id" : "(optional, string) API ID of the Canvas variation this event belongs to",
  "device_id" : "(optional, string) ID of the device on which the event occurred",
  "dispatch_id" : "(optional, string) ID of the dispatch this message belongs to",
  "external_user_id" : "(optional, string) External ID of the user",
  "from_phone_number" : "(required, string) Phone number used to send in e.164 format (for example +14155552671)",
  "id" : "(required, string) Globally unique ID for this event",
  "message_variation_id" : "(optional, string) API ID of the message variation this user received",
  "provider_error_code" : "(required, string) Error code from WhatsApp",
  "provider_error_title" : "(required, string) Description of the error from WhatsApp",
  "subscription_group_id" : "(optional, string) Subscription group API ID",
  "time" : "(required, long) Unix timestamp at which the event happened",
  "timezone" : "(optional, string) Time zone of the user",
  "to_phone_number" : "(optional, string) Phone number of the user receiving the message in e.164 format (for example +14155552671)",
  "user_id" : "(required, string) BSON ID of the user who performed this event"
```

File: _docs/_user_guide/data_and_analytics/braze_currents/event_glossary/message_engagement_events.md [hunk 2]
```diff
@@ -119,1786 +70,9439 @@ WhatsApp, Delivery
...
-## Webhook abort message events
+## WhatsApp abort message events
 
 {% apitags %}
-Abort,  Webhooks
+WhatsApp, Abort
 {% endapitags %}
 
-This event occurs if a webhook message was aborted based on Liquid aborts, Quiet Hours, etc.
+This event occurs if a WhatsApp message was aborted based on Liquid aborts, Quiet Hours, etc.
```
Comment (lydia-xie) [inline _docs/_user_guide/data_and_analytics/braze_currents/event_glossary/message_engagement_events.md:713]:
```suggestion
This event occurs if a WhatsApp message is aborted based on Liquid aborts or Quiet Hours.
```

**Review by lydia-xie — COMMENTED** [review]
Hey @isturdy ! Overall, the PR looks good to me! I have a few edits but they're pretty duplicative across the events:

- `External ID of the user` instead of `External user ID of the user`
- `e.164 format` instead of `e164 format`
- `Description of error from WhatsApp` instead of `Description of failure from WhatsApp`
- `ID of the card this in-app message comes from` instead of `ID of the card this in app message comes from`

There are also some extra periods in the field descriptions that we can remove as well from the Mixpanel events

File: _docs/_user_guide/data_and_analytics/braze_currents/event_glossary/customer_behavior_events.md [hunk 3]
```diff
@@ -53,26 +52,114 @@ Custom Events
...
-  "device_model": (optional, string) hardware model of the device,
-  "device_id": (optional, string) ID of the device on which the event occurred,
-  "name": (required, string) name of the custom event,
-  "properties": (required, string) JSON encoded string of the properties for this event,
-  "ad_id": (optional, string) advertising identifier,
-  "ad_id_type": (optional, string) One of 'ios_idfa', 'google_ad_id', OR 'roku_ad_id',
-  "ad_tracking_enabled": (optional, boolean) whether advertising tracking is enabled for the device
+  "event" : "(required, string) The event type name, as it is exported to Mixpanel.",
+  "properties" : {
+    "\\$partner_id" : "braze",
```
[the remark this answers is not in the document]
Comment (isturdy) [reply _docs/_user_guide/data_and_analytics/braze_currents/event_glossary/customer_behavior_events.md:93]: No--I don't know if it's historical or to meet partner requirements, but this is exported in all lowercase while the Amplitude equivalent is capitalized.

File: _docs/_user_guide/data_and_analytics/braze_currents/event_glossary/customer_behavior_events.md [hunk 4]
```diff
@@ -40,8 +40,7 @@ Certain events return a `platform` value that specifies the platform of the user
 {% enddetails %}
 
 {% alert important %}
-Note that these schemas only apply to the flat file event data we send to Data Warehouse partners (Google Cloud Storage, Amazon S3, and Microsoft Azure Blob Storage), and are not available for Segment connectors. For schema that apply to other partners, refer to our list of [available partners]({{site.baseurl}}/user_guide/data_and_analytics/braze_currents/available_partners/) and check their respective pages.<br><br>Additionally, note that Currents will drop events with excessively large payloads of greater than 900&nbsp;KB. 
-
+"Storage" schemas apply to the flat file event data we send to Data Warehouse partners (Google Cloud Storage, Amazon S3, and Microsoft Azure Blob Storage). Some event/destination combinations listed here are not yet generally available; for information on which events are supported by various partners, refer to our list of [available partners]({{site.baseurl}}/user_guide/data_and_analytics/braze_currents/available_partners/) and check their respective pages.<br><br>Additionally, note that Currents will drop events with excessively large payloads of greater than 900&nbsp;KB.
```
Comment (rachel-feinberg) [inline _docs/_user_guide/data_and_analytics/braze_currents/event_glossary/customer_behavior_events.md:43]:
```suggestion
Storage schemas apply to the flat file event data we send to data warehouse partners (Google Cloud Storage, Amazon S3, and Microsoft Azure Blob Storage). Some event and destination combinations listed here are not yet generally available. For information on which events are supported by various partners, refer to our list of [available partners]({{site.baseurl}}/user_guide/data_and_analytics/braze_currents/available_partners/) and check their respective pages.<br><br>Additionally, note that Currents will drop events with excessively large payloads of greater than 900&nbsp;KB.
```

**Review by rachel-feinberg — APPROVED** [review]
Lgtm!

**Pull request merged**